commit 1407a9e105d811632b3fd57741a1b3d72a26c9e6
Author: Larry Finger <Larry.Finger@lwfinger.net>
Date:   Mon Aug 30 07:51:09 2010 -0500

    staging: r8712u: Fix rate setting when HT is not enabled
    
    When operating in normal bg mode, the maximum rate was fixed at 48 Mb/s.
    
    Signed-off-by: Albert Wang <albert_wang@realtek.com.tw>
    Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>

--- a/drivers/staging/rtl8712/rtl871x_ioctl_linux.c
+++ b/drivers/staging/rtl8712/rtl871x_ioctl_linux.c
@@ -1371,6 +1371,8 @@ static int r8711_wx_get_rate(struct net_
 					    135) : ((short_GI) ? 72 : 65);
 			max_rate *= 2; /* Mbps/2 */
 			wrqu->bitrate.value = max_rate * 500000;
+		} else {
+			wrqu->bitrate.value = max_rate * 500000;
 		}
 	} else
 		return -1;
